### DISCUSSION TITLE
Applying update versions of leaf to a production environment.

### DISCUSSION THREAD

**Discussion by glabrie10** [uwrit/leaf#494] [Q&A] [answered]
Hello group,

Currently, we at HDC would like to update our leaf instance,
Here are the proposed steps:

1. Pull updated version of leaf from the source repository.
2. Re-run: npm run build.  This will create a new build bundle, to be copied to the webserver
3. Recopy new build folder to the webserver /etc/www.
4. restart application and webserver

I'm wondering if there are additional steps needed to update a leaf build in production?

best,
Gabe

**Comment by mh2727** [comment, marked as answer]
Hi Gabe.

To update with the latest Leaf code on GitHub, you'll also have to run the appropriate database migration scripts and rebuild the API on the server.

This workflow is needed to get the latest features and fixes from the Leaf source code.

Steps to update Leaf from source code:

1. Pull updated version of leaf from the source repository.
2. Run: `npm run build`. This will create a new build bundle, to be copied to the webserver.
3. Copy new build folder to the webserver (`/etc/www` directory or where your web directory is located).
4. Run the appropriate database migration scripts.
5. Rebuild the API on the server.
6. Start application and webserver.

Hope this clarifies the steps.

--
Mehadi Hassan

**Reply by glabrie10** [reply]
@mh2727 Could you clarify where #4(Run the appropriate database migration scripts) is in the documentation?

**Reply by mh2727** [reply]
@glabrie10,

Under releases on the Leaf GitHub repo, there's documentation on the latest updates/fixes along with the necessary files to make the changes.

Here's the link to it: https://github.com/uwrit/leaf/releases. The steps are outlined under "Upgrading to ...".

--
Mehadi Hassan

**Reply by glabrie10** [reply]
Thank you!

**Comment by glabrie10** [comment]
I consider this answered!
